### PR TITLE
Add smt::Context

### PR DESCRIPTION
**Pull request by strikef**
This PR adds `smt::Context` and related functions to toggle options

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -112,6 +112,8 @@ int main(int argc, char* argv[]) {
   llvm::cl::ParseCommandLineOptions(argc, argv);
 
   z3::set_param("timeout", (int)arg_smt_to.getValue());
```
Comment (aqjune) [inline src/main.cpp:114]: Remove this?

File: src/smt.cpp [hunk 2]
```diff
@@ -20,25 +21,40 @@
...
+    IF_Z3_ENABLED(this->z3.reset());
+    IF_CVC5_ENABLED(this->cvc5.reset());
+    timeout_ms = 20000;
+  }
+
+  IF_Z3_ENABLED(void useZ3() { this->z3.emplace(); })
+  IF_CVC5_ENABLED(void useCVC5() { this->cvc5.emplace(); })
 
-Context sctx(true);
+  string getFreshName(string prefix) {
```
Comment (aqjune) [inline src/smt.cpp:50]: Is this related to Context?

File: src/smt.cpp [hunk 3]
```diff
@@ -622,8 +638,69 @@ Model Solver::getModel() const {
...
+
+  z3::expr arr(*sctx.z3, Z3_get_app_arg(*sctx.z3, a, 0));
+  z3::expr idx(*sctx.z3, Z3_get_app_arg(*sctx.z3, a, 1));
+  z3::expr val(*sctx.z3, Z3_get_app_arg(*sctx.z3, a, 2));
+
+  return arrMatcher(createExpr(move(arr))) &&
+      idxMatcher(createExpr(move(idx))) &&
+      valMatcher(createExpr(move(val)));
+}
+}
```
Comment (aqjune) [inline src/smt.cpp:685]: // namespace matchers

File: CMakeLists.txt [hunk 4]
```diff
@@ -97,5 +96,4 @@ target_compile_options(mlir-interp PRIVATE -fno-rtti)
 
 enable_testing()
 add_subdirectory(${PROJECT_SOURCE_DIR}/tests)
-# Reactivate this after unit tests are updated to use the new SMT wrapper classes
-#add_subdirectory(${PROJECT_SOURCE_DIR}/unittests)
+add_subdirectory(${PROJECT_SOURCE_DIR}/unittests)
```
Comment (aqjune) [inline CMakeLists.txt:99]: commentize

File: src/smt.cpp (hunk 2 above)
Comment (strikef) [reply src/smt.cpp:51]: CVC5 does not natively support fresh naming, so use our own semantics
